src/app/_components/Avatar.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+
+import Image from 'next/image'
+
+import { type DigestArticleData } from '@/types/digestTypes'
+import type { ImageProps } from '@/types/sharedProps/imageType'
+
+type ImagePropsWithoutAlt = Omit<ImageProps, 'alt'>
+
+export function Avatar({ authors }: { authors: DigestArticleData['authors'] }) {
+  const imageSize = 32
+
+  if (!authors || authors.length === 0) {
+    throw new Error('Avatar component requires at least one author')
+  }
+
+  function renderImage(
+    firstName: string,
+    lastName: string,
+    image: ImagePropsWithoutAlt,
+  ) {
+    return (
+      <Image
+        className="inline-block rounded-full object-cover ring-2 ring-brand-700 ring-opacity-30"
+        src={image.src}
+        alt={`Photo of ${firstName} ${lastName}`}
+        width={imageSize}
+        height={imageSize}
+      />
+    )
+  }
+
+  function renderInitials(firstName: string, lastName: string) {
+    const initials = (firstName[0] + lastName[0]).toUpperCase()
+    return (
+      <div
+        style={{ width: imageSize, height: imageSize }}
+        className="flex items-center justify-center rounded-full bg-brand-700 ring-2 ring-brand-700 ring-opacity-30"
+        aria-label={`Initials of ${firstName} ${lastName}`}
+      >
+        <span className="text-sm font-medium leading-none text-brand-300">
+          {initials}
+        </span>
+      </div>
+    )
+  }
+
+  function renderAuthorContent({
+    firstName,
+    lastName,
+    image,
+  }: DigestArticleData['authors'][number]) {
+    return image?.src
+      ? renderImage(firstName, lastName, image)
+      : renderInitials(firstName, lastName)
+  }
```

Comment:
Just curious as to why these are functions and not components?

Reply:
I find it that they're specific to the Avatar logic and aren't meant to be reused elsewhere.